packages/legend-studio/src/components/editor/edit-panel/GrammarTextEditor.tsx

```diff
@@ -91,6 +91,7 @@ export const GrammarTextEditor = observer(() => {
         ...baseTextEditorSettings,
         language: EDITOR_LANGUAGE.PURE,
         theme: EDITOR_THEME.STUDIO,
+        readOnly: editorStore.isInViewerMode,
```

Comment:
Let's not set this here, but instead in the `if (editor)` block, because this `useEffect` will only be called once when we initialize the editor